src/irr/asset/CMeshManipulator.cpp

```diff
@@ -1551,6 +1551,343 @@ core::smart_refctd_ptr<ICPUBuffer> IMeshManipulator::idxBufferFromTrianglesFanTo
+    core::vectorSIMDf AABBmid; //axis aligned mid point of the vertices
+    float AABBArea;        //quality measure of the axis-aligned box 
+
+    core::vectorSIMDf p0, p1, p2; // Vertices of the large base triangle
+    core::vectorSIMDf e0, e1, e2; // Edge vectors of the large base triangle
+    core::vectorSIMDf n;          // Unit normal of the large base triangle 
+    core::vectorSIMDf t0, t1;     // tetrahedron vertices
+
+    core::matrix3x4SIMD b; // The currently best found OBB orientation (transposed)
+    float bestVal;         // the best obb quality value
+    core::vectorSIMDf bMin, bMax, bLen;
+
+    constexpr float eps = 0.000001f;
+
+    const core::vectorSIMDf sampleDir[projCnt] =
+    {
+        core::vectorSIMDf(1.0f, 0.0f, 0.0f),
+        core::vectorSIMDf(0.0f, 1.0f, 0.0f),
+        core::vectorSIMDf(0.0f, 0.0f, 1.0f),
+        core::vectorSIMDf(1.0f, 1.0f, 1.0f),
+        core::vectorSIMDf(1.0f, 1.0f, -1.0f),
+        core::vectorSIMDf(1.0f, -1.0f, 1.0f),
+        core::vectorSIMDf(1.0f, -1.0f, -1.0f),
+        core::vectorSIMDf(1.0f, 1.0f, 0.0f),
+        core::vectorSIMDf(1.0f, -1.0f, 0.0f),
+        core::vectorSIMDf(1.0f, 0.0f, 1.0f),
+        core::vectorSIMDf(1.0f, 0.0f, -1.0f),
+        core::vectorSIMDf(0.0f, 1.0f, 1.0f),
+        core::vectorSIMDf(1.0f, 1.0f, -1.0f)
+    };
+
+    auto constructOBB = [](const core::matrix3x4SIMD& rotation, const core::vectorSIMDf& scale, const core::vectorSIMDf& mid)
+    {
+        core::matrix3x4SIMD scaleMat;
+        scaleMat.setScale(scale);
+
+        IMeshManipulator::OBB result; 
+        result.asMat3x4 = core::concatenateBFollowedByA(rotation, scaleMat);
+        result.asMat3x4.setTranslation(mid);
+
+        return result;
+    };
+    
+    //calc max and min projections
+    {
+        const core::vectorSIMDf firstVtxPos = mb->getPosition(0u);
+
+        for(uint32_t i = 0u; i < projCnt; i++)
+            minProjLen[i] = maxProjLen[i] = core::dot(firstVtxPos, sampleDir[i]).x; // should be better to compute it manually..
+
+        std::fill(minVtxPos, minVtxPos + projCnt, firstVtxPos);
+        std::fill(maxVtxPos, maxVtxPos + projCnt, firstVtxPos);
+
+        for (size_t i = 1u; i < mb->calcVertexCount(); i++)
+        {
+            for (uint32_t j = 0u; j < projCnt; j++)
+            {
+                float vtxProj = core::dot(mb->getPosition(i), sampleDir[j]).x;
+
+                if (vtxProj > maxProjLen[j])
+                {
+                    maxProjLen[j] = vtxProj;
+                    maxVtxPos[j] = mb->getPosition(i);
+                }
+
+                if (vtxProj < minProjLen[j])
+                {
+                    minProjLen[j] = vtxProj;
+                    minVtxPos[j] = mb->getPosition(i);
+                }
+            }
+        }
+    }
+
+    //compute size of AABB (slabs 0, 1 and 2 define AABB)
+    AABBmid = core::vectorSIMDf(minProjLen[0] + maxProjLen[0], minProjLen[1] + maxProjLen[1], minProjLen[2] + maxProjLen[2]) * 0.5f;
```

Comment:
I wouldn't keep track of the center, I'd keep track of origin (OBB forms an orthogonal local basis with scale)